File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/LICHLAMVIEC.cs

using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class LICHLAMVIEC : BaseEntity
    {
        [Key]
        public Guid MaLLV { get; set; }

        public DateTime NgayLamViec { get; set; }
        public DateTime GioBatDau { get; set; }
        public DateTime GioKetThuc { get; set; }

        [ForeignKey("BacSi")]
        public Guid? MaBS { get; set; }
        public BACSI? BacSi { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/LichLamViec.js
/**
 * @typedef {Object} LichLamViec
 * @property {string} maLLV
 * @property {string|Date} ngayLamViec
 * @property {string|Date} gioBatDau
 * @property {string|Date} gioKetThuc
 * @property {string} maBS
 * @property {string} tenBacSi
 */

export const createLichLamViec = (
  maLLV = "",
  ngayLamViec = null,
  gioBatDau = null,
  gioKetThuc = null,
  maBS = null,
  tenBacSi = "",
) => ({
  maLLV,
  ngayLamViec,
  gioBatDau,
  gioKetThuc,
  maBS,
  tenBacSi,
});

export const normalizeLichLamViec = (item) => ({
  maLLV: item?.maLLV ?? item?.MaLLV ?? "",
  ngayLamViec: item?.ngayLamViec ?? item?.NgayLamViec ?? null,
  gioBatDau: item?.gioBatDau ?? item?.GioBatDau ?? null,
  gioKetThuc: item?.gioKetThuc ?? item?.GioKetThuc ?? null,
  maBS: item?.maBS ?? item?.MaBS ?? null,
  tenBacSi: item?.tenBacSi ?? item?.TenBacSi ?? "",
});
